subgraphs/uniswap-v3/src/mappings/core.ts

```diff
@@ -1,72 +1,216 @@
+    event.params.amount,
+    activeLiquidityDelta,
+    [BIGINT_ZERO, BIGINT_ZERO],
+    [BIGINT_ZERO, BIGINT_ZERO]
+  );
+  dexEventHandler.createDeposit(
     event.params.owner,
-    event.params.amount0,
-    event.params.amount1
+    getOrCreateTick(event, pool, BigInt.fromI32(event.params.tickLower)),
+    getOrCreateTick(event, pool, BigInt.fromI32(event.params.tickUpper)),
+    null
+  );
+  dexEventHandler.processLPBalanceChanges();
+  incrementDepositHelper(dexEventHandler.pool.id);
+}
+
+// Handle burn event emmitted from a pool contract.
+export function handleBurn(event: BurnEvent): void {
+  const pool = getLiquidityPool(event.address)!;
+  let activeLiquidityDelta = BIGINT_ZERO;
+  // Make sure the liquidity is within the current tick range to update active liquidity.
+  if (
+    pool.tick !== null &&
+    BigInt.fromI32(event.params.tickLower).le(pool.tick as BigInt) &&
+    BigInt.fromI32(event.params.tickUpper).gt(pool.tick as BigInt)
+  ) {
+    activeLiquidityDelta = event.params.amount.times(BIGINT_NEG_ONE);
+  }
+  pool._totalAmountWithdrawn = sumBigIntListByIndex([
+    pool._totalAmountWithdrawn,
+    [event.params.amount0, event.params.amount1],
+  ]);
+  const dexEventHandler = new DexEventHandler(
+    event,
+    event.transaction.from,
+    pool,
+    false,
+    [
+      event.params.amount0.times(BIGINT_NEG_ONE),
+      event.params.amount1.times(BIGINT_NEG_ONE),
+    ],
+    event.params.amount.times(BIGINT_NEG_ONE),
+    activeLiquidityDelta,
+    [BIGINT_ZERO, BIGINT_ZERO],
+    [BIGINT_ZERO, BIGINT_ZERO]
   );
-  updateUsageMetrics(event, event.params.owner, UsageType.DEPOSIT);
-  updateFinancials(event);
-  updatePoolMetrics(event);
+  dexEventHandler.createWithdraw(
+    event.params.owner,
+    getOrCreateTick(event, pool, BigInt.fromI32(event.params.tickLower)),
+    getOrCreateTick(event, pool, BigInt.fromI32(event.params.tickUpper)),
+    null
+  );
+  dexEventHandler.processLPBalanceChanges();
 }
 
 // Handle a swap event emitted from a pool contract.
 export function handleSwap(event: SwapEvent): void {
-  createSwapHandleVolumeAndFees(
-    event,
+  updateTokenPrices(event, event.params.sqrtPriceX96);
+  const pool = getLiquidityPool(event.address)!;
+  const swapHelperObj = getSwapHelperObj(
+    pool,
+    event.params.liquidity,
     event.params.amount0,
-    event.params.amount1,
-    event.params.recipient,
+    event.params.amount1
+  );
+  pool._totalAmountEarned = sumBigIntListByIndex([
+    pool._totalAmountEarned,
+    swapHelperObj.protocolUncollectedTokenChangesNet,
+    swapHelperObj.supplyUncollectedTokenChangesNet,
+  ]);
+  const dexEventHandler = new DexEventHandler(
+    event,
     event.transaction.from,
-    event.params.sqrtPriceX96
+    pool,
+    true,
+    swapHelperObj.balanceChangesNet,
+    BIGINT_ZERO,
+    swapHelperObj.activeLiquidityDelta,
+    swapHelperObj.supplyUncollectedTokenChangesNet,
+    swapHelperObj.protocolUncollectedTokenChangesNet
```

Comment:
Does it make sense to have `DexEventHandler` have `SwapHelperObj` as an argument instead of its fields? This way it is easier for a user (of DexEventHandler) to know what to pass to it. Also reduces the chances of passing the fields swapped.

Or is it used in more cases than just Swaps? If this is the case, instead of instantiating it with the constructor you might have different functions (static maybe) that initialize the `DexEventHandler` in different ways, depending on what might be available at each time: `Deposit, Swap, Withdraw, etc`

Reply:
So I have it set up so that the constructor takes a set of deltas. Then the deltas are propagated to update all the fields in the entities. What do you think about creating a function for each that instantiates the same `Deltas` type, and then pass that along for the last 5 args? Or one function for Deposits + Withdraws and one for Swaps.

Reply:
I like passing the deltas, because the constructor initializes more or less the same based on the deltas. The only thing that is different is that the deltas count for volume with swaps, which is why I pass the flag. wdyt?

Reply:
>  What do you think about creating a function for each that instantiates the same Deltas type, and then pass that along for the last 5 args?

Yes, I like that. That way you also encapsulate the logic that creates the deltas from a given event type. Which could eventually be generalized.